docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -103,33 +103,89 @@ to install it:
 ```dotnetcli
 > dotnet tool update -g dotnet-counters
 You can invoke the tool using the following command: dotnet-counters
-Tool 'dotnet-counters' (version '5.0.251802') was successfully installed.
-```
-
-While the example app is still running, list the running processes in a second shell to determine the process ID:
-
-```dotnetcli
-> dotnet-counters ps
-     10180 dotnet     C:\Program Files\dotnet\dotnet.exe
-     19964 metric-instr E:\temp\metric-instr\bin\Debug\netcoreapp3.1\metric-instr.exe
+Tool 'dotnet-counters' (version '7.0.430602') was successfully installed.
 ```
 
-Find the ID for the process name that matches the example app and have dotnet-counters monitor the new counter:
+While the example app is still running, use dotnet-counters to monitor the new counter:
 
 ```dotnetcli
-> dotnet-counters monitor -p 19964 HatCo.HatStore
+> dotnet-counters monitor -n metric-demo.exe --counters HatCo.Store
 Press p to pause, r to resume, q to quit.
     Status: Running
 
-[HatCo.HatStore]
-    hats-sold (Count / 1 sec)                          4
+[HatCo.Store]
+    hatco.store.hats_sold (Count / 1 sec)                          4
 ```
 
 As expected, you can see that HatCo store is steadily selling 4 hats each second.
 
+## Getting a Meter via Dependency Injection
+
+In the example above we obtained the Meter by constructing it with `new` and assigning it to a static field. Using statics this way is not a good approach when using Dependency
+Injection (DI). In code that uses DI such as ASP.NET Core, or apps with [Generic Host](../extensions/generic-host.md) create the Meter object using
+<xref:System.Diagnostics.Metrics.IMeterFactory>. Starting in .NET 8 hosts will automatically register <xref:System.Diagnostics.Metrics.IMeterFactory> in the service container
+or you can manually register the type in any <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection> by calling <xref:Microsoft.Extensions.DependencyInjection.MetricsServiceExtensions.AddMetrics%2A>.
+The meter factory integrates metrics with DI, keeping Meters in different service collections isolated from each other even if they are using an identical name. This is
+especially useful for testing so that multiple tests running in parallel will only observe measurements produced from within the same test case.
```

Comment:
```suggestion
especially useful for testing so that multiple tests running in parallel only observe measurements produced from within the same test case.
```